pkg/engine/engine.go

```diff
@@ -595,14 +595,29 @@ func getAddonFuncMap(addon api.KubernetesAddon, cs *api.ContainerService) templa
 		"HasAvailabilityZones": func() bool {
 			return cs.Properties.HasAvailabilityZones()
 		},
-		"GetZones": func() string {
+		"HasAgentPoolAvailabilityZones": func() bool {
+			for _, pool := range cs.Properties.AgentPoolProfiles {
+				if pool.AvailabilityZones != nil {
+					return true
+				}
+			}
+			return false
+		},
+		"GetAgentPoolZones": func() string {
 			if len(cs.Properties.AgentPoolProfiles) == 0 {
 				return ""
 			}
 
 			var zones string
-			for _, zone := range cs.Properties.AgentPoolProfiles[0].AvailabilityZones {
-				zones += fmt.Sprintf("\n    - %s-%s", cs.Location, zone)
+			for _, pool := range cs.Properties.AgentPoolProfiles {
+				if pool.AvailabilityZones != nil {
+					for _, zone := range cs.Properties.AgentPoolProfiles[0].AvailabilityZones {
```

Comment:
```suggestion
					for _, zone := range pool.AvailabilityZones {
```

Reply:
Also, `zone` will have duplicated entries if there is more than 1 agent pool with the same availability zone. Is that the intention here?

Reply:
It's not the intention. The if zones != "" then return zones is supposed to short-circuit the func so that the first pool w/ zones will return those as the value

Reply:
Ah, I see that now.

Reply:
fixed the static reference to the first pool, thanks!